src/ProxyManager/GeneratorStrategy/FileWriterGeneratorStrategy.php

```diff
@@ -74,9 +73,12 @@ public function generate(ClassGenerator $classGenerator) : string
      */
     private function writeFile(string $source, string $location) : void
     {
-        $tmpFileName = tempnam($location, 'temporaryProxyManagerFile');
+        $directory   = dirname($location);
+        $tmpFileName = tempnam($directory, 'temporaryProxyManagerFile');
 
-        assert(is_string($tmpFileName));
+        if ($tmpFileName === false) {
+            throw FileNotWritableException::fromNotWritableDirectory($directory);
```

Comment:
Is there a way to hit this code location?

Reply:
Yes, see https://github.com/Ocramius/ProxyManager/pull/438#discussion_r241368819